src/Globals.h

```diff
@@ -200,46 +200,43 @@ template class SizeChecker<UInt8,  1>;
 // Common headers (part 1, without macros):
 #include "fmt.h"
 #include "StringUtils.h"
+#include "LoggerSimple.h"
 #include "OSSupport/CriticalSection.h"
 #include "OSSupport/Event.h"
 #include "OSSupport/File.h"
 #include "OSSupport/StackTrace.h"
 
-#ifndef TEST_GLOBALS
-
-	#include "LoggerSimple.h"
+#ifdef TEST_GLOBALS
 
-#else
-	#include "fmt/printf.h"
+	// Basic logging function implementations
+	namespace Logger
+	{
 
-	// Logging functions
-	template <typename ... Args>
-	void LOG(const char * a_Format, const Args & ... a_Args)
+	inline void LogFormat(
+		std::string_view a_Format, eLogLevel, fmt::format_args a_ArgList
+	)
 	{
-		fmt::printf(a_Format, a_Args...);
+		fmt::vprint(a_Format, a_ArgList);
 		putchar('\n');
 		fflush(stdout);
 	}
 
-	#define LOGERROR   LOG
-	#define LOGWARNING LOG
-	#define LOGD       LOG
-	#define LOGINFO    LOG
-	#define LOGWARN    LOG
-
-	template <typename ... Args>
-	void FLOG(const char * a_Format, const Args & ... a_Args)
+	inline void LogPrintf(
+		std::string_view a_Format, eLogLevel, fmt::printf_args a_ArgList
+	)
 	{
-		fmt::print(a_Format, a_Args...);
+		fmt::vprintf(a_Format, a_ArgList);
 		putchar('\n');
 		fflush(stdout);
 	}
 
-	#define FLOGERROR   FLOG
-	#define FLOGWARNING FLOG
-	#define FLOGD       FLOG
-	#define FLOGINFO    FLOG
-	#define FLOGWARN    FLOG
+	inline void LogSimple(std::string_view a_Message, eLogLevel)
+	{
+		fmt::print("{}\n", a_Message);
```

Comment:
std::cout << a_Message << std::endl ?

Reply:
Why though? This works the same and saves an include.

Reply:
There's an outstanding issue for migrating to streams for output.

Reply:
Nah, there was an issue for modernising the logging. Which is why `fmt` was added. iostreams are not the way to go IMO.

Reply:
I'm referring to #799

what do you mean they're not the way to go, this is C++...

Reply:
Nonetheless, it seems out of scope for this PR.

Reply:
Well iostreams lack i18n support which xoft mentioned in #3542. They're also  notoriously slow since they have their own buffering on top of the system's standard buffers.